### PR TITLE
[posix] fix unused variable when log level is less than OT_LOG_LEVEL_CRIT

### PR DESCRIPTION
**Pull request by suveshpratapa**
Encountered this problem while building POSIX stack with logging enabled.

### CODE REVIEW

File: src/posix/platform/misc.cpp [hunk 1]
```diff
@@ -78,7 +78,12 @@ otError otPlatSetMcuPowerState(otInstance *aInstance, otPlatMcuPowerState aState
 
 void otPlatAssertFail(const char *aFilename, int aLineNumber)
 {
+#if OPENTHREAD_CONFIG_LOG_LEVEL < OT_LOG_LEVEL_CRIT
```
Comment (Irving-cl) [inline src/posix/platform/misc.cpp:81]:
```suggestion
#if OPENTHREAD_CONFIG_LOG_PLATFORM && OPENTHREAD_CONFIG_LOG_LEVEL < OT_LOG_LEVEL_CRIT
```

Comment (suveshpratapa) [reply src/posix/platform/misc.cpp]: Fixed! Thank you.

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**